wal/src/manager.rs

```diff
@@ -286,44 +278,41 @@ pub trait WalManager: Send + Sync + fmt::Debug + 'static {
     async fn scan(&self, ctx: &ScanContext, req: &ScanRequest) -> Result<BatchLogIteratorAdapter>;
 }
 
-/// Adapter to convert a blocking interator to a batch async iterator.
+#[derive(Debug)]
+pub enum InnerIterator {
+    Sync(Box<dyn SyncLogIterator>, Arc<Runtime>),
+    Async(Box<dyn AsyncLogIterator>),
+}
+
+/// Adapter to convert a sync/async iterator to a batch async iterator.
 #[derive(Debug)]
 pub struct BatchLogIteratorAdapter {
-    blocking_iter: Option<Box<dyn BlockingLogIterator>>,
-    runtime: Arc<Runtime>,
+    inner_iterator: Option<InnerIterator>,
     batch_size: usize,
 }
 
 impl BatchLogIteratorAdapter {
-    pub fn new(
-        blocking_iter: Box<dyn BlockingLogIterator>,
-        runtime: Arc<Runtime>,
-        batch_size: usize,
-    ) -> Self {
+    pub fn new(inner_iterator: InnerIterator, batch_size: usize) -> Self {
```

Comment:
Maybe we should define two methods: `new_sync` & `new_async`.

Reply:
Yes, quite verbose to new it now.

Reply:
Done.